trunk/Lib/Sidebar.php

```diff
@@ -0,0 +1,695 @@
+     * @param string $version The version of this plugin.
+     * @since    1.0.0
+     */
+    public function __construct( $pluginName, $version ) {
+
+        $this->pluginName = $pluginName;
+        $this->version = $version;
+
+        $this->loader = new Loader();
+
+        $this->defineHooks();
+
+        $this->loader->run();
+
+        $this->categoryList = get_categories();
+    }
+
+    /**
+     * Register all of the hooks related to the sidebar functionality
+     */
+    private function defineHooks() {
+        $this->loader->addAction( 'wp_enqueue_scripts', $this, 'enqueueStyles' );
+        $this->loader->addAction( 'wp_enqueue_scripts', $this, 'enqueueScripts' );
+        $this->loader->addAction( 'admin_enqueue_scripts', $this, 'enqueueStyles' );
+        $this->loader->addAction( 'admin_enqueue_scripts', $this, 'enqueueScripts' );
+        $this->loader->addAction( 'elementor/editor/before_enqueue_styles', $this, 'enqueueStyles' );
+        $this->loader->addAction( 'elementor/editor/before_enqueue_scripts', $this, 'enqueueScripts', 99999 );
+
+        // embed to footer on admin
+        $this->loader->addAction( 'admin_print_footer_scripts', $this, 'renderSidebar' );
+
+        // embed to footer on website
+        $this->loader->addAction( 'wp_footer', $this, 'renderSidebar' );
+
+        $this->loader->addAction( 'set_current_user', $this, 'fixElementorLanguage', 11 );
+        $this->loader->addAction( 'wp_ajax_aqbRenamePost', $this, 'renamePost' );
+
+    }
+
+    /**
+     * Checks if wmpl plugin is active
+     *
+     * @return bool
+     */
+    private function isWpmlActive() {
+        return is_plugin_active( 'the_wplm_plugin_slug' );
+    }
+
+
+    /**
+     * Set post-types
+     */
+    public function setPostTypes() {
+        $this->postTypes = get_post_types( [], 'object' );
+    }
+
+    /**
+     * Set filtered post-types
+     */
+    public function setFilteredPostTypes() {
+        foreach ( $this->postTypes as $postType ) {
+            if ( in_array( $postType->name, $this->filterPostTypes ) ) {
+                continue;
+            }
+            $this->filteredPostTypes[] = $postType;
+        }
+    }
+
+    /**
+     * Get cache list from swift and writes to $cacheList
+     */
+    public function initCacheList() {
+        if ( !$this->hasSwift() ) {
+            return;
+        }
+
+        $this->cacheList = class_exists( 'Swift_Performance' )
+            ? Swift_Performance::cache_status()['files']
+            : Swift_Performance_Lite::cache_status()['files'];
+    }
+
+    private function hasSwift() {
+        return class_exists( 'Swift_Performance' ) || class_exists( 'Swift_Performance_Lite' );
+    }
+
+    /**
+     * Adds the sidebar to footer
+     *
+     * @param string $data
+     * @return string $data
+     * @throws \ImagickException
+     */
+    public function renderSidebar( $data ) {
+        $this->isWpmlActive();
```

Comment:
This line doesn't currently do anything.

Reply:
Yes, is removed